src/slot.rs

```diff
@@ -822,7 +854,29 @@ impl<Conn: Connection> SetWorker<Conn> {
     async fn run(&mut self) {
         loop {
             tokio::select! {
+                biased;
+                // If we should exit, terminate immediately
                 _ = &mut self.terminate_rx => {
+                    while let Some((_id, slot)) = self.slots.pop_first() {
+                        let handle = {
+                            let mut lock = slot.inner.guarded.lock().unwrap();
+
+                            // First, fire the oneshot, so the background task
+                            // starts to exit.
+                            if let Some(tx) = lock.terminate_tx.take() {
+                                let _send_result = tx.send(());
+                            }
+
+                            // Next, pull out the handle, so we can watch it
+                            // terminate.
+                            let Some(handle) = lock.handle.take() else {
+                                return;
+                            };
+                            handle
+                        };
+
+                        handle.await.expect("Slot worker panicked");
+                    }
```

Comment:
So, I notice that there's also a `return` later in this select in the `request = self.rx.recv()` branch, where the request channel has been dropped. We should probably also tear down all the slots in that case, right?

Perhaps we should change this branch of the select as well as that one to just `break` out of the loop, and put this code outside the loop, so that it happens in either case?

Reply:
Oh, shoot, I see that you just merged the branch --- I can open a PR to make that change if you like?

Reply:
Appreciate the feedback -- I"m making these changes now!

Reply:
Oop, you beat me to it in https://github.com/oxidecomputer/qorb/pull/47. Looks good!